src/server.c

```diff
@@ -4009,6 +4009,17 @@ int processCommand(client *c) {
         }
     }
 
+    if (!server.cluster_enabled &&
+        server.masterhost &&
+        !mustObeyClient(c) &&
+        server.repl_replica_enable_redirect &&
```

Comment:
Let's move the line repl_replica_enable_redirect a little ahead (next to cluster_enabled)? This is similar to a big switch on the front. I think it would be better to read simple judgment if it is placed on the front.